### PR TITLE
soc: nordic: Minor updates for nRF54H20

### PR DESCRIPTION
**Pull request by 57300**

### CODE REVIEW

**Review by anangl — DISMISSED** [review]

**Review by gmarull — DISMISSED** [review]

**Review by nordic-krch — DISMISSED** [review]

**Comment by nashif** [comment]
needs rebase

**Comment by aescolar** [comment]
@57300 please remember to rebase

**Comment by aescolar** [comment]
@57300 . To keep up with the tradition, this is your weekly "please remember to rebase this PR" reminder.

**Comment by 57300** [comment]
> I'm setting this PR as dnm until it is rebased, so it stops being in the short merge list.

Sorry @aescolar , I wasn't aware this was making noise. Rebased now.

**Review by anangl — APPROVED** [review]

File: soc/nordic/validate_base_addresses.c [hunk 1]
```diff
@@ -37,6 +37,14 @@
 #define NRF_QDEC0 NRF_QDEC
 #endif
 
+#if !defined(NRF_RADIO) && defined(NRF_RADIOCORE_RADIO)
+#define NRF_RADIO NRF_RADIOCORE_RADIO
```
Comment (masz-nordic) [inline soc/nordic/validate_base_addresses.c:41]: The root cause, in my understanding, is that we are trying to verify peripherals from radio domain when building for other cores.
Not a blocker for me, but perhaps a cleaner solution would be to have those peripherals:
https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/common/nordic/nrf54h20.dtsi#L173-L296
only included in radio core builds?
Unless this is needed in some use cases that I'm not aware of.

Comment (57300) [reply soc/nordic/validate_base_addresses.c:41]: Pretty much. The root cause is that local peripherals are also visible to the secure domain, but it's not present upstream, so I tried to justify this patch in another way.

**Review by masz-nordic — APPROVED** [review]

**Pull request merged**